Tests fail when using pytest with SQLite and TranslationSource.get_or_create_from_instance

wagtail = "4.2"
wagtail-localize = "^1.4"

We are running tests using `pytest` with the `--nomigrations` flag and SQLite as the db backend.

When running tests involving `TranslationSource.get_or_create_from_instance`, the tests fail with the following error:

```
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/wagtail_localize/models.py:381: in get_or_create_from_instance
    "schema_version": get_schema_version(instance._meta.app_label) or "",
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/wagtail_localize/models.py:136: in get_schema_version
    .last()
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/cacheops/query.py:363: in last
    return self._no_monkey.last(self)
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/models/query.py:1055: in last
    for obj in (self.reverse() if self.ordered else self.order_by("-pk"))[:1]:
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/models/query.py:394: in __iter__
    self._fetch_all()
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/cacheops/query.py:273: in _fetch_all
    return self._no_monkey._fetch_all(self)
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/models/query.py:1867: in _fetch_all
    self._result_cache = list(self._iterable_class(self))
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/models/query.py:87: in __iter__
    results = compiler.execute_sql(
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/models/sql/compiler.py:1398: in execute_sql
    cursor.execute(sql, params)
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/cacheops/transaction.py:98: in execute
    result = self._no_monkey.execute(self, sql, params)
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/backends/utils.py:67: in execute
    return self._execute_with_wrappers(
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/backends/utils.py:80: in _execute_with_wrappers
    return executor(sql, params, many, context)
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/backends/utils.py:84: in _execute
    with self.db.wrap_database_errors:
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/utils.py:91: in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
/root/.cache/pypoetry/virtualenvs/.venv/lib/python3.10/site-packages/django/db/backends/utils.py:89: in _execute
    return self.cursor.execute(sql, params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.sqlite3.base.SQLiteCursorWrapper object at 0xffffb63171c0>
query = 'SELECT "django_migrations"."id", "django_migrations"."app", "django_migrations"."name", "django_migrations"."applied" FROM "django_migrations" WHERE "django_migrations"."app" = ? ORDER BY "django_migrations"."applied" DESC LIMIT 1'
params = ('core',)

    def execute(self, query, params=None):
        if params is None:
            return Database.Cursor.execute(self, query)
        query = self.convert_query(query)
>       return Database.Cursor.execute(self, query, params)
E       django.db.utils.OperationalError: no such table: django_migrations
```

This looks like it's because `MigrationRecorder` is being used here: 

https://github.com/wagtail/wagtail-localize/blob/v1.4/wagtail_localize/models.py#L139

Is there a way around this for testing `TranslationSource` using `pytest` and `SQLite`?

--- a/wagtail_localize/models.py
+++ b/wagtail_localize/models.py
@@ -8,7 +8,7 @@ from django.contrib.admin.utils import quote
 from django.contrib.contenttypes.models import ContentType
 from django.core.exceptions import ValidationError
 from django.core.serializers.json import DjangoJSONEncoder
-from django.db import models, transaction
+from django.db import OperationalError, models, transaction
 from django.db.migrations.recorder import MigrationRecorder
 from django.db.models import (
     Case,
@@ -126,17 +126,21 @@ def get_edit_url(instance):
         )
 
 
-def get_schema_version(app_label):
+def get_schema_version(app_label: str) -> str:
     """
     Returns the name of the last applied migration for the given app label.
     """
-    migration = (
-        MigrationRecorder.Migration.objects.filter(app=app_label)
-        .order_by("applied")
-        .last()
-    )
-    if migration:
-        return migration.name
+    try:
+        if migration := (
+            MigrationRecorder.Migration.objects.filter(app=app_label)
+            .order_by("applied")
+            .last()
+        ):
+            return migration.name
+    except OperationalError:
+        ...
+
+    return ""
 
 
 class TranslatableObjectManager(models.Manager):
@@ -378,7 +382,7 @@ class TranslationSource(models.Model):
                 "locale": instance.locale,
                 "object_repr": str(instance)[:200],
                 "content_json": content_json,
-                "schema_version": get_schema_version(instance._meta.app_label) or "",
+                "schema_version": get_schema_version(instance._meta.app_label),
                 "last_updated_at": timezone.now(),
             },
         )
@@ -436,7 +440,7 @@ class TranslationSource(models.Model):
                 "locale": instance.locale,
                 "object_repr": str(instance)[:200],
                 "content_json": content_json,
-                "schema_version": get_schema_version(instance._meta.app_label) or "",
+                "schema_version": get_schema_version(instance._meta.app_label),
                 "last_updated_at": timezone.now(),
             },
         )
@@ -460,7 +464,7 @@ class TranslationSource(models.Model):
             serializable_data = get_serializable_data_for_fields(instance)
             self.content_json = json.dumps(serializable_data, cls=DjangoJSONEncoder)
 
-        self.schema_version = get_schema_version(instance._meta.app_label) or ""
+        self.schema_version = get_schema_version(instance._meta.app_label)
         self.object_repr = str(instance)[:200]
         self.last_updated_at = timezone.now()
 

--- a/wagtail_localize/tests/test_translation_model.py
+++ b/wagtail_localize/tests/test_translation_model.py
@@ -2,6 +2,9 @@ from unittest import mock
 
 import polib
 
+from django.conf import settings
+from django.db import OperationalError
+from django.db.migrations.recorder import MigrationRecorder
 from django.test import TestCase, override_settings
 from django.utils import timezone
 from wagtail.models import Locale, Page
@@ -23,6 +26,7 @@ from wagtail_localize.models import (
     TranslationSource,
     UnknownContext,
     UnknownString,
+    get_schema_version,
 )
 from wagtail_localize.segments import RelatedObjectSegmentValue
 from wagtail_localize.strings import StringValue
@@ -1108,3 +1112,30 @@ class TestTranslationSourceViewRestrictions(TestCase):
 
         with self.assertRaises(NoViewRestrictionsError):
             source.sync_view_restrictions(snippet, snippet)
+
+
+class TestHelpers(TestCase):
+    def test_get_schema_version_with_no_migrations(self):
+        self.assertEqual(get_schema_version("test.TestSnippet"), "")
+
+    def test_get_schema_version_with_a_migration(self):
+        self.assertEqual(get_schema_version("test.TestSnippet"), "")
+        MigrationRecorder.Migration.objects.create(
+            app="test.TestSnippet",
+            name="0001_initial",
+        )
+        self.assertEqual(get_schema_version("test.TestSnippet"), "0001_initial")
+
+    @mock.patch("wagtail_localize.models.MigrationRecorder.Migration.objects.filter")
+    def test_get_schema_version_returns_empty_string_on_operational_error(
+        self, mocked_filter
+    ):
+        mocked_filter.side_effect = OperationalError
+
+        self.assertEqual(get_schema_version("test.TestSnippet"), "")
+
+    def test_get_schema_version_with_TEST_MIGRATE_false(self):
+        databases = settings.DATABASES.copy()
+        databases["TEST"] = {"MIGRATE": False}
+        with override_settings(DATABASES=databases):
+            self.assertEqual(get_schema_version("test.TestSnippet"), "")

